tests/phpunit/tests/compat/array_key_first.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * @group compat
+ * @ticket 45055
+ *
+ * @covers ::array_key_first
+ */
+class Tests_Compat_array_key_first extends WP_UnitTestCase {
+
+	/**
+	 * Test that array_key_first() is always available (either from PHP or WP).
+	 */
+	public function test_array_key_first_availability() {
+		$this->assertTrue( function_exists( 'array_key_first' ) );
+	}
+
+	/**
+	 * @dataProvider array_key_first_provider
+	 */
+	public function test_array_key_first( $expected, $array ) {
+		if ( ! function_exists( 'array_key_first' ) ) {
+			$this->markTestSkipped( 'array_key_first() is not available.' );
+		} else {
+			$this->assertSame(
+				$expected,
+				array_key_first( $array )
+			);
+		}
+
+	}
+
+	/**
+	 * Data provider for test_array_key_first().
+	 *
+	 * @return array[]
+	 */
+	public function array_key_first_provider() {
```

Comment:
```suggestion
	/**
	 * @ticket 45055
	 * @dataProvider data_array_key_first
	 * @param int|string|null $expected The expected result.
	 * @param array $array The array.
	 */
	public function test_array_key_first( $expected, $array ) {
		if ( ! function_exists( 'array_key_first' ) ) {
			$this->markTestSkipped( 'array_key_first() is not available.' );
		} else {
			$this->assertSame(
				$expected,
				array_key_first( $array )
			);
		}

	}

	/**
	 * Data provider for test_array_key_first().
	 *
	 * @return array
	 */
	public function data_array_key_first() {
```

- All tests should state their ticket.
- Suggest renaming this data provider to prefix it with `data_array_key_first` per [Trac 53119](https://core.trac.wordpress.org/ticket/53119).
- For tests with data providers, all parameters should be documented.
- `@return array[]` doesn't need the `[]`.